sql/src/main/java/org/apache/druid/sql/calcite/rel/DruidQuery.java

```diff
@@ -897,6 +906,11 @@ public GroupByQuery toGroupByQuery()
       return null;
     }
 
+    if (sorting != null && sorting.getOffsetLimit().hasLimit() && sorting.getOffsetLimit().getLimit() <= 0) {
```

Comment:
can limit be less than 0 here because of the preconditions in `OffsetLimit`?

Reply:
It can't, but I guess it doesn't hurt to overcheck.